src/software/ai/hl/stp/tactic/attacker/attacker_fsm.h

```diff
@@ -0,0 +1,121 @@
+        std::optional<Point> chip_target;
+        // shoot goal config
+        std::shared_ptr<const AttackerTacticConfig> attacker_tactic_config;
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto pivot_kick_s = state<PivotKickFSM>;
+        const auto keep_away_s  = state<DribbleFSM>;
+        const auto update_e     = event<Update>;
+
+        /**
+         * Action that updates the PivotKickFSM to shoot or pass
+         *
+         * @param event AttackerFSM::Update event
+         * @param processEvent processes the PivotKickFSM::Update
+         */
+        const auto pivot_kick = [](auto event,
+                                   back::process<PivotKickFSM::Update> processEvent) {
+            auto ball_position = event.common.world.ball().position();
+            Point chip_target  = event.common.world.field().enemyGoalCenter();
+            if (event.control_params.chip_target)
+            {
+                chip_target = event.control_params.chip_target.value();
+            }
+            // default to chiping the ball away
+            PivotKickFSM::ControlParams control_params{
+                .kick_origin    = ball_position,
+                .kick_direction = (chip_target - ball_position).orientation(),
+                .auto_chip_or_kick =
+                    AutoChipOrKick{AutoChipOrKickMode::AUTOCHIP,
+                                   (chip_target - ball_position).length()}};
+
+            if (event.control_params.shot)
+            {
+                // shoot on net
+                control_params = PivotKickFSM::ControlParams{
+                    .kick_origin = ball_position,
+                    .kick_direction =
+                        (event.control_params.shot->getPointToShootAt() - ball_position)
+                            .orientation(),
+                    .auto_chip_or_kick =
+                        AutoChipOrKick{AutoChipOrKickMode::AUTOKICK,
+                                       BALL_MAX_SPEED_METERS_PER_SECOND - 0.5}};
+            }
+            else if (event.control_params.pass)
+            {
+                control_params = PivotKickFSM::ControlParams{
+                    .kick_origin    = event.control_params.pass->passerPoint(),
+                    .kick_direction = event.control_params.pass->passerOrientation(),
+                    .auto_chip_or_kick =
+                        AutoChipOrKick{AutoChipOrKickMode::AUTOKICK,
+                                       event.control_params.pass->speed()}};
+            }
+            processEvent(PivotKickFSM::Update(control_params, event.common));
+        };
+
+        /**
+         * Action that updates the DribbleFSM to chip away the ball
+         *
+         * @param event AttackerFSM::Update event
+         * @param processEvent processes the DribbleFSM::Update
+         */
+        const auto keep_away = [](auto event,
```

Comment:
Does this just make the robot stand still with the ball, since there's no destination/orientation?

Reply:
yeah, added a TODO